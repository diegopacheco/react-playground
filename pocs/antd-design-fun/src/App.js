import './App.css';
import { Button, DatePicker } from 'antd';

function App() {
  return (
    <div className="App">
      <main>
        <Button type="primary">PRESS ME</Button>
        <DatePicker placeholder="select date" />
      </main>
    </div>
  );
}

export default App;
