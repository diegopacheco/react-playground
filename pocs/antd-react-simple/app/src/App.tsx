import React from 'react';
import logo from './logo.svg';
import './App.css';
import { DatePicker } from 'antd';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <DatePicker />
      </header>
    </div>
  );
}

export default App;
