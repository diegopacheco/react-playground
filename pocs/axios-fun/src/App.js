import logo from './logo.svg';
import './App.css';
import PersonList from './PersonList';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        Axios
      </header>
      <main>
          <PersonList />
      </main>
    </div>
  );
}

export default App;
