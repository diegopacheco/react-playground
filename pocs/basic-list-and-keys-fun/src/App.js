import logo from './logo.svg';
import './App.css';

function TodoList(props) {
  const todos = props.todos;
  const todoItems = todos.map((todo) =>
    <li>{todo}</li>
  );
  return (
    <div>
      <p>TODO Items:</p>
      <ul>{todoItems}</ul>
    </div>
  );
}

const numbers = [1, 2, 3, 4, 5];
const listItems = numbers.map((number) =>
  <li>{number}</li>
)

export default function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        Lists and Keys
      </header>
      <main>
        <div>
          Numbers:
          <ul>{listItems}</ul>
         </div>
         <TodoList todos={["Learn React","Learn TS","Rule the Universe"]} />
      </main>
    </div>
  );
}