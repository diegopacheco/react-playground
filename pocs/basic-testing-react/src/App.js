import logo from './logo.svg';
import './App.css';
import Hello from './Hello';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        Basic for Testing
      </header>
      <main>
        <br/>
        <Hello></Hello>
      </main>
    </div>
  );
}

export default App;
