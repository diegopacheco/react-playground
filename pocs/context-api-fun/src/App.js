import logo from './logo.svg';
import './App.css';
import {ThemeContext, themes} from './theme-context';
import ThemedButton from './themed-button';
import React from 'react';
import Page from './Page';

function Toolbar(props) {
  return (
    <ThemedButton onClick={props.changeTheme}>
      Change Theme
    </ThemedButton>
  );
}

class App extends React.Component{
  constructor(props) {
    super(props);
    this.state = {
      theme: themes.dark,
    };

    this.toggleTheme = () => {
      this.setState(state => ({
        theme:
          state.theme === themes.dark
            ? themes.light
            : themes.dark,
      }));
    };
  }
  render(){
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          Context Api
        </header>
        <main>
        <ThemeContext.Provider value={this.state.theme}>
            <Page>
              <Toolbar changeTheme={this.toggleTheme} />
              <div>
                <ThemedButton />
              </div>
            </Page>
         </ThemeContext.Provider>         
        </main>
      </div>
    ); 
  }
}

export default App;
