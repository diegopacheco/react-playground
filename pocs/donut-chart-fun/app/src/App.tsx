import React from 'react';
import logo from './logo.svg';
import './App.css';
import DonutChart from 'react-donut-chart';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <DonutChart
          strokeColor="black"
          emptyColor="gray"
          data={[
            {
              label: 'Give you up',
              value: 25,
            },
            {
              label: '',
              value: 75,
              isEmpty: true,
            },
          ]}
        />;
      </header>
    </div>
  );
}

export default App;
