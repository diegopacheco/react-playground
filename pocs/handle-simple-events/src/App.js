import logo from './logo.svg';
import './App.css';
import React from 'react';

class Toggle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isToggleOn: true};
    // This binding is necessary to make `this` work in the callback
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.setState(state => ({
      isToggleOn: !state.isToggleOn
    }));
  }
  render() {
    return (
      <button onClick={this.handleClick}>
        {this.state.isToggleOn ? 'ON' : 'OFF'}
      </button>
    );
  }
}

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        Handle Events
      </header>
      <main>
        <br/>  
        <Toggle />
      </main>
    </div>
  );
}

export default App;
