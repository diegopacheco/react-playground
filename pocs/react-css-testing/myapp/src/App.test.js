import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('[CSS-TEST] h1 has underline class', () => {
  render(<App />);
  const h1Element = screen.getByText(/Hello world from tailwind!/i);

  // checking all CSS
  expect(h1Element).toHaveClass('underline');
  expect(h1Element).toHaveClass('text-3xl');
  expect(h1Element).toHaveClass('font-bold');
});
