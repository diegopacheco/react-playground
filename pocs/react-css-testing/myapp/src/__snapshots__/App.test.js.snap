// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <br />
      <h1
        class="text-3xl font-bold underline"
      >
        Hello world from tailwind!
      </h1>
    </header>
  </div>
</DocumentFragment>
`;
